packages/esm-patient-notes-app/src/notes/visit-notes-form.component.tsx

```diff
@@ -244,61 +307,78 @@ const VisitNotesForm: React.FC<DefaultWorkspaceProps> = ({ closeWorkspace, patie
           </Column>
           <Column sm={3}>
             <div className={styles.diagnosesText} style={{ marginBottom: '1.188rem' }}>
-              {selectedDiagnoses && selectedDiagnoses.length ? (
+              {selectedPrimaryDiagnoses && selectedPrimaryDiagnoses.length ? (
+                <>
+                  {selectedPrimaryDiagnoses.map((diagnosis, index) => (
+                    <Tag
+                      filter
+                      key={index}
+                      onClose={() => handleRemoveDiagnosis(diagnosis, 'primaryInputSearch')}
+                      style={{ marginRight: '0.5rem' }}
+                      type={'red'}
+                    >
+                      {diagnosis.display}
+                    </Tag>
+                  ))}
+                </>
+              ) : (
+                <></>
+              )}
+              {selectedSecondaryDiagnoses && selectedSecondaryDiagnoses.length ? (
                 <>
-                  {selectedDiagnoses.map((diagnosis, index) => (
+                  {selectedSecondaryDiagnoses.map((diagnosis, index) => (
                     <Tag
                       filter
                       key={index}
-                      onClose={() => handleRemoveDiagnosis(diagnosis)}
+                      onClose={() => handleRemoveDiagnosis(diagnosis, 'secondaryInputSearch')}
                       style={{ marginRight: '0.5rem' }}
-                      type={index === 0 ? 'red' : 'blue'}
+                      type={'blue'}
                     >
-                      {diagnosis.concept.preferredName}
+                      {diagnosis.display}
                     </Tag>
                   ))}
                 </>
               ) : (
+                <></>
+              )}
+              {selectedPrimaryDiagnoses &&
+              !selectedPrimaryDiagnoses.length &&
+              selectedSecondaryDiagnoses &&
+              !selectedSecondaryDiagnoses.length ? (
                 <span>{t('emptyDiagnosisText', 'No diagnosis selected — Enter a diagnosis below')}</span>
+              ) : (
+                <></>
               )}
             </div>
-            <FormGroup legendText={t('searchForDiagnosis', 'Search for a diagnosis')}>
+            <FormGroup legendText={t('searchForPrimaryDiagnosis', 'Search for a primary diagnosis')}>
               <Search
-                light={isTablet}
                 size="lg"
-                id="diagnosisSearch"
-                labelText={t('enterDiagnoses', 'Enter diagnoses')}
-                placeholder={t(
-                  'diagnosisInputPlaceholder',
-                  'Choose a primary diagnosis first, then secondary diagnoses',
-                )}
-                onChange={handleSearchTermChange}
+                id="diagnosisPrimarySearch"
+                labelText={t('enterPrimaryDiagnoses', 'Enter Primary diagnosis')}
```

Comment:
The translation key implies this is a plural:

```suggestion
                labelText={t('enterPrimaryDiagnoses', 'Enter Primary diagnoses')}
```